hplot/s2d.go

```diff
@@ -100,16 +103,28 @@ func (pts *S2D) withBand() error {
 			bot[i].X = x
 			bot[i].Y = y - math.Abs(ymin)
 		}
+
 	case HiSteps:
-		for i := range top {
-			// WIP(rmadar): implement the proper band in case of step
-			//              this might actually involve 2xn points - need to think
-			x, y := pts.Data.XY(i)
+		top = make(plotter.XYs, 2*pts.Data.Len())
+		bot = make(plotter.XYs, 2*pts.Data.Len())
+		xerr, ok := pts.Data.(plotter.XErrorer)
+		if !ok {
+			panic("s2d: cannot get X errors, required for HiSteps plotting")
```

Comment:
perhaps this check could be performed earlier?
(in `NewS2D`)

Reply:
Actually this is not strictly speaking need: the way the step are done will still work without X errors, but will give the same result as NoSteps. So I would tend to remove this `panic` ... What do you think?

Here is an example for those points:
```go
pts := []hbook.Point2D{                                                                            
          {X: 1, Y: 1, ErrY: hbook.Range{Min: 2, Max: 3}},                                           
          {X: 2, ErrX: hbook.Range{Min: 0.5, Max: 0.5}, Y: 2, ErrY: hbook.Range{Min: 5, Max: 2}},    
          {X: 3, Y: 3, ErrY: hbook.Range{Min: 2, Max: 2}},                                           
          {X: 4, ErrX: hbook.Range{Min: 0.5, Max: 0.5}, Y: 4, ErrY: hbook.Range{Min: 1.2, Max: 2}},  
}
```

![image](https://user-images.githubusercontent.com/30081646/80291106-f9699f80-874a-11ea-8b71-02e29811839c.png)

Reply:
sure, no panics is better :)